src/code/z_demo.c

```diff
@@ -1,99 +1,1634 @@
 #include "global.h"
+#include "overlays/gamestates/ovl_daytelop/z_daytelop.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/Cutscene_Init.s")
+void Cutscene_DoNothing(GlobalContext* globalCtx, CutsceneContext* csCtx);
+void func_800EA258(GlobalContext* globalCtx, CutsceneContext* csCtx);
+void func_800ED9C4(GlobalContext* globalCtx, CutsceneContext* csCtx);
+void func_800EA2B8(GlobalContext* globalCtx, CutsceneContext* csCtx);
+void func_800ED980(GlobalContext* globalCtx, CutsceneContext* csCtx);
+void func_800EDA04(GlobalContext* globalCtx, CutsceneContext* csCtx);
+void func_800EDA84(GlobalContext* globalCtx, CutsceneContext* csCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EA0D4.s")
+// Unused
+UNK_TYPE4 D_801BB120 = 0;
+u16 D_801BB124 = 0;
+u16 D_801BB128 = 0;
+u8 D_801BB12C = 0;
+u8 sCutsceneStoredPlayerForm = 0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EA0EC.s")
+// bss
+#ifndef NON_MATCHING
+static u16 seqId;
+#endif
+s16 sCutsceneQuakeIndex;
+struct_801F4D48 sCutsceneCameraInfo;
+u16 D_801F4DC8[10];
+UNK_TYPE D_801F4DDC;
+u8 D_801F4DE0;
+s16 D_801F4DE2;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/Cutscene_StepCutscene1.s")
+void Cutscene_Init(GlobalContext* globalCtx, CutsceneContext* csCtx) {
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/Cutscene_StepCutscene2.s")
+    csCtx->state = CS_STATE_0;
+    csCtx->frames = 0;
+    csCtx->unk_0C = 0.0f;
+    globalCtx->csCtx.sceneCsCount = 0;
+    globalCtx->csCtx.currentCsIndex = 0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/Cutscene_Nop800EA210.s")
+    for (i = 0; i < ARRAY_COUNT(D_801F4DC8); i++) {
+        D_801F4DC8[i] = 0;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EA220.s")
+    D_801F4DE0 = 0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EA258.s")
+    Audio_SetCutsceneFlag(false);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EA2B8.s")
+void Cutscene_Start(GlobalContext* globalCtx, CutsceneContext* csCtx) {
+    csCtx->state = CS_STATE_1;
+    csCtx->playerAction = NULL;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EA324.s")
+void Cutscene_End(GlobalContext* globalCtx, CutsceneContext* csCtx) {
+    if (csCtx->state != CS_STATE_4) {
+        csCtx->state = CS_STATE_3;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EABAC.s")
+typedef void (*CutsceneStateHandler)(GlobalContext* globalCtx, CutsceneContext* csCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EAC08.s")
+CutsceneStateHandler sCsStateHandlers1[] = {
+    /* CS_STATE_0 */ Cutscene_DoNothing,
+    /* CS_STATE_1 */ func_800EA258,
+    /* CS_STATE_2 */ Cutscene_DoNothing,
+    /* CS_STATE_3 */ func_800ED9C4,
+    /* CS_STATE_4 */ Cutscene_DoNothing,
+};
```

Comment:
I think the style we've been going for is to use `// ` comments on the right side